### ISSUE TITLE
Linux support

### ISSUE THREAD

**Issue by daviddgm** [irusanov/ZenTimings#18]
Hi! Do you have plan for Linux version? Or do you have code to read memories timings in Linux?

**Comment by irusanov** [comment]
I want to add linux support, since that's also my main OS, but currently it's not easy to port it. It's not cross-platform.

**Comment by daviddgm** [comment]
OK. I found this https://github.com/leogx9r/ryzen_smu and maybe useful when you will port.

**Comment by irusanov** [comment]
Yes, I was going to mention it. It has support for some of the timings.

**Comment by jbosboom** [comment]
[hattedsquirrel/ryzen_monitor](https://github.com/hattedsquirrel/ryzen_monitor)'s `-m` switch prints most of the timings ZenTimings displays, and its main interface includes some of the voltages.

**Comment by AdamKafei** [comment]
Is there any life in this idea still?

**Comment by jfroy** [comment]
I have **an extremely janky prototype** available at https://github.com/jfroy/ddrs based on the information in this projects and others. I have no intention of developing it beyond a useful prototype, but it can serve as a reference for someone else interested in an actual production-quality tool.

**Comment by irusanov** [comment]
There's another linux project that looks promising: https://github.com/Death4two/TuxTimings

**Comment by Marc-Pierre-Barbier** [comment]
<img width="410" height="754" alt="Image" src="https://github.com/user-attachments/assets/2f8db0be-d38c-4365-b032-f12d26e61be5" />

I was working on my own ZenTimings clone for linux.

I have a lot of issues trying to read from the ACPI table (AODE). when parsing the AODE header i get an offset actually bigger than the table.

I get something like that:

```
41 4f 44 45       #name 
00                      #region space
0c                      #unknown 
18 80 f5 72      #Offset ~411104626
0b                     # unknown 2
bb 24               # length
5b 81 40          # unkown 3 to 5
```
The table I'm reading from is /sys/firmware/acpi/tables/SSDT8 and is only 10437B big. i would love to have some advice from @irusanov

**Comment by irusanov** [comment]
Hi, what you have as offset from the region is in fact a physical memory address where the AOD table is copied. It's a SystemMemory region.

Here's an example of what rweverything parses the AML to:
`OperationRegion(AODT, SystemMemory, 0xCCBEC018, 0x220C)`

First number is address and the second one is the size.
Maybe you can take a look at the other linux project and how it is handled there.

**Comment by Marc-Pierre-Barbier** [comment]
From reading your message and the code for Tux Clocker I now believe it's only possible to access this memory through a kernel module, as arbitrary memory addresses can only be accessed with kernel level privileges.

I made it to be a relatively small project easy to use. I don't feel like adding a kernel module that tux clocker already use.

I will strip all of my ACPI code and publish the code in a gh repository.
